boa_engine/src/builtins/intl/mod.rs

```diff
@@ -171,9 +206,11 @@ fn lookup_matcher(
     for locale_str in requested_locales {
         // a. Let noExtensionsLocale be the String value that is locale with any Unicode locale
         //    extension sequences removed.
-        let parsed_locale =
-            Locale::from_bytes(locale_str.as_bytes()).expect("Locale parsing failed");
-        let no_extensions_locale = JsString::new(parsed_locale.id.to_string());
+        let maybe_locale = Locale::from_bytes(locale_str.as_bytes());
+        let no_extensions_locale = match &maybe_locale {
+            Ok(parsed_locale) => JsString::new(parsed_locale.id.to_string()),
+            Err(_) => trim_unicode_extensions(locale_str),
```

Comment:
`suggestion`: if it is an invalid locale, then you shouldn't take it at all.

Reply:
Done. https://github.com/boa-dev/boa/pull/2072/files#diff-1dd4af22d3a6142a8060d056e9d0489d84d8a4d9c68a8f6e7b359b8d3ab35499R167